zerver/tests/test_realm.py

```diff
@@ -1594,7 +1595,13 @@ def do_test_realm_permission_group_setting_update_api_with_anonymous_groups(
         realm = get_realm("zulip")
         othello = self.example_user("othello")
         hamlet = self.example_user("hamlet")
-        leadership_group = check_add_user_group(realm, "leadership", [hamlet], acting_user=None)
+
+        try:
+            leadership_group = check_add_user_group(realm, "leadership", [hamlet], acting_user=None)
+        except JsonableError:
+            leadership_group = NamedUserGroup.objects.get(
+                name="leadership", realm=realm, is_system_group=False
+            )
```

Comment:
when the method is being called by multiple settings, then `check_add_user_group` would throw error since group already present in db.

Reply:
updated!